### PR TITLE
Register append function to list type

### PR DESCRIPTION
**Pull request by windelbouwman**
This is my attempt to add append to the list object

### CODE REVIEW

File: vm/src/pyobject.rs [hunk 1]
```diff
@@ -262,6 +264,7 @@ impl AttributeProtocol for PyObjectRef {
             PyObjectKind::Module { name: _, ref dict } => dict.contains_key(attr_name),
             PyObjectKind::Class { name: _, ref dict } => dict.contains_key(attr_name),
             PyObjectKind::Instance { ref dict } => dict.contains_key(attr_name),
+            PyObjectKind::List { elements: _ } => self.typ().has_attr(attr_name),
```
Comment (cthulahoops) [inline vm/src/pyobject.rs]: ```has_attr``` should return false. The list itself doesn't have attributes (only its class does).

File: vm/src/pyobject.rs [hunk 2]
```diff
@@ -252,6 +253,7 @@ impl AttributeProtocol for PyObjectRef {
             PyObjectKind::Module { name: _, ref dict } => dict.get_item(attr_name),
             PyObjectKind::Class { name: _, ref dict } => dict.get_item(attr_name),
             PyObjectKind::Instance { ref dict } => dict.get_item(attr_name),
+            PyObjectKind::List { elements: _ } => self.typ().get_attr(attr_name),
```
Comment (cthulahoops) [inline vm/src/pyobject.rs]: Should raise ```AttributeError``` or possibly panic for the moment until this code returns a result.

File: vm/src/pyobject.rs (hunk 1 above)
Comment (windelbouwman) [reply vm/src/pyobject.rs]: Agreed!

File: vm/src/pyobject.rs (hunk 2 above)
Comment (windelbouwman) [reply vm/src/pyobject.rs]: Agreed, reverted this line so it will panic.

**Comment by cthulahoops** [comment]
I see what you mean about the chicken-egg problem with all these methods. I think we need a vm.new_method to handle it.

This looks good though.

**Pull request merged**